src/pyseekdb/utils/embedding_functions/onnx_embedding_function.py

```
@@ -0,0 +1,392 @@
+
+                    self._download(url, local_path, chunk_size=8192)
+                    logger.info(f"Successfully downloaded {local_filename}")
+                except httpx.HTTPStatusError as e:
+                    if e.response.status_code == 404:
+                        logger.warning(f"File {hf_filename} not found on Hugging Face (404), will try fallback")
+                        return False
+                    logger.warning(f"HTTP error downloading {hf_filename} from Hugging Face: {e}")
+                    if os.path.exists(local_path):
+                        os.remove(local_path)
+                    return False
+                except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)